pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -1209,52 +1206,80 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
-			ServiceInformer:                informerFactory.Core().V1().Services(),
-			PdbInformer:                    informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
-			StorageClassInformer:           informerFactory.Storage().V1().StorageClasses(),
-			CSINodeInformer:                informerFactory.Storage().V1beta1().CSINodes(),
-			HardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
-			DisablePreemption:              false,
-			PercentageOfNodesToScore:       schedulerapi.DefaultPercentageOfNodesToScore,
-		}).CreateFromConfig(policy); err != nil {
+		sched, err := scheduler.New(
+			client,
+			informerFactory.Core().V1().Nodes(),
+			informerFactory.Core().V1().Pods(),
+			informerFactory.Core().V1().PersistentVolumes(),
+			informerFactory.Core().V1().PersistentVolumeClaims(),
+			informerFactory.Core().V1().ReplicationControllers(),
+			informerFactory.Apps().V1().ReplicaSets(),
+			informerFactory.Apps().V1().StatefulSets(),
+			informerFactory.Core().V1().Services(),
+			informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
+			informerFactory.Storage().V1().StorageClasses(),
+			informerFactory.Storage().V1beta1().CSINodes(),
+			nil,
+			algorithmSrc,
+			make(chan struct{}),
+			schedulerframework.NewDefaultRegistry(),
+			nil,
+			[]kubeschedulerconfig.PluginConfig{},
+		)
+		if err != nil {
 			t.Errorf("%s: Error constructing: %v", v, err)
 			continue
 		}
-	}
-
-	if !seenPredicates.HasAll(registeredPredicates.List()...) {
-		t.Errorf("Registered predicates are missing from compatibility test (add to test stanza for version currently in development): %#v", registeredPredicates.Difference(seenPredicates).List())
-	}
-	if !seenPriorities.HasAll(registeredPriorities.List()...) {
-		t.Errorf("Registered priorities are missing from compatibility test (add to test stanza for version currently in development): %#v", registeredPriorities.Difference(seenPriorities).List())
+		schedPredicates := sets.NewString()
+		for p := range sched.Algorithm.Predicates() {
+			schedPredicates.Insert(p)
+		}
+		wantPredicates := sets.NewString()
+		wantPredicates.Insert("CheckNodeCondition") // mandatory predicate
+		for _, p := range tc.ExpectedPolicy.Predicates {
+			wantPredicates.Insert(p.Name)
+		}
+		schedPrioritizers := sets.NewString()
+		for _, p := range sched.Algorithm.Prioritizers() {
+			schedPrioritizers.Insert(p.Name)
+		}
+		wantPrioritizers := sets.NewString()
+		for _, p := range tc.ExpectedPolicy.Priorities {
+			wantPrioritizers.Insert(p.Name)
+		}
+		if !schedPredicates.Equal(wantPredicates) {
+			t.Errorf("Expected predicates %v, got %v", wantPredicates, schedPredicates)
+		}
+		if !schedPrioritizers.Equal(wantPrioritizers) {
+			t.Errorf("Expected prioritizers %v, got %v", wantPrioritizers, schedPrioritizers)
+		}
+		var wantAlgoExtenders []algorithm.SchedulerExtender
+		schedExtenders := sched.Algorithm.Extenders()
+		for _, e := range tc.ExpectedPolicy.ExtenderConfigs {
```

Comment:
@alculquicondor  Couldn't figure out the right way to compare extenders. The following did not work. Any thoughts?

Reply:
from here: https://github.com/kubernetes/kubernetes/blob/a0c93b10c9173476d284ac5a7cd8569bc91dedff/pkg/scheduler/factory/factory.go#L347

it looks like all extenders should be of the same type `HTTPExtender` you can have `wantExtenders []*HttpExtender` that you can compare.

Reply:
`core.NewHTTPExtender` returns an algorithm.SchedulerExtender. Why would we use HTTPExtender then?

Reply:
The point is that all of them are of the same type. And if you want to compare their configuration, that might be the only option.